Initialization of Spring SecurityContext for non HTTP requests

nenaraab opened the pull request:
In case of non-HTTP requests, you may need to initialize the Spring SecurityContext with a JWT token you've received from a message or you've requested from XSUAA directly.

**Prerequisite**:   
You need to initialize a `JwtDecoder` bean, as explained [here](https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/samples/spring-security-xsuaa-usage/src/main/java/sample/spring/xsuaa/SecurityConfiguration.java), using the `XsuaaJwtDecoderBuilder` class.

**Then, initialize Security Context**:
```
@Autowired
JwtDecoder jwtDecoder;

public void initializeSecurityContext() {
    Jwt jwtToken = jwtDecoder.decode(<your encoded jwt token String>);
    SecurityContext.init(XSAPPNAME, jwtToken, true);
}
```

By default the SecurityContext is initialized for your current Thread. I.e. you can access the Token from everywhere in your code, that runs in the same Thread:
```
Token token = (Token)SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

**Finally, clear Security Context**:
Note that `SecurityContextHolder` stores the information in `ThreadLocal`s. That means you should remove current thread's value at the end:
```
SecurityContext.clear();
```

nenaraab commented:
@mwdb 
Hi Martijn, can you please review?

mwdb commented:
Hi,
Please add documentation + sample + CHANGELOG do this PR.

Regards,

Martijn

nenaraab commented:
@mwdb : requested changes applied

- updated changelog
- spring-xsuaa-it
  - added sample: [InitializeSecurityContextTest](https://github.com/SAP/cloud-security-xsuaa-integration/blob/aebf658a8911491b3f9b56616620b7730f51144e/spring-xsuaa-it/src/test/java/testservice/api/InitializeSecurityContextTest.java) which uses [MyEventHandler](https://github.com/SAP/cloud-security-xsuaa-integration/blob/aebf658a8911491b3f9b56616620b7730f51144e/spring-xsuaa-it/src/main/java/testservice/api/nohttp/MyEventHandler.java)
  - increased version in pom.xml, cleanup
- [Readme.md](https://github.com/SAP/cloud-security-xsuaa-integration/blob/aebf658a8911491b3f9b56616620b7730f51144e/spring-xsuaa/README.md)
